fsm.go

```diff
@@ -118,7 +118,7 @@ func (fsm *FSM) Restore(reader io.ReadCloser) error {
 
 	fsm.mux.Lock()
 	defer fsm.mux.Unlock()
-	if err := DecodeSnapshot(snapBytes, &fsm.stateWrap); err != nil {
+	if err := DecodeSnapshot(snapBytes, fsm.stateWrap.State); err != nil {
```

Comment:
try passing a pointer?

Reply:
Unfortunately passing a pointer leads to  "This object does not implement marshable" when working with ipfs-cluster.  Interestingly enough it also doesn't suffice to fix restoration in the transport test.